cf-check/dump.c

```diff
@@ -438,25 +439,17 @@ static bool matches_option(
 int dump_main(int argc, const char *const *const argv)
 {
     assert(argv != NULL);
-    if (argc < 2) // Need at least binary and 1 filename
-    {
-        print_usage();
-        return EXIT_FAILURE;
-    }
+    assert(argc >= 1);
+
     dump_mode mode = DUMP_MODE_NICE;
     const char *const *filenames = argv + 1;
     size_t filenames_len = argc - 1;
-    if (argv[1][0] == '-')
+    if (filenames_len > 0 && filenames[0] != NULL && filenames[0][0] == '-')
```

Comment:
`filenames[0][0] == '-'` is really confusing. What about just checking `argv` here and define `filenames` and `filenames_len` when it's clear where the file names start?

Reply:
I considered both, and thought this was cleaner (less thinking about indices / offsets), but then I'll change it.